addons/website_event/static/src/js/website_event.editor.js

```diff
@@ -2,11 +2,201 @@ odoo.define('website_event.editor', function (require) {
 "use strict";
 
 var core = require('web.core');
-var wUtils = require('website.utils');
+var time = require('web.time');
 var WebsiteNewMenu = require('website.newMenu');
-
+var Dialog = require('web.Dialog');
 var _t = core._t;
 
+var EventCreateDialog = Dialog.extend({
+    xmlDependencies: Dialog.prototype.xmlDependencies.concat(
+        [
+            '/website_event/static/src/xml/website_event_templates.xml'
+        ]
+    ),
+    template: 'website_event.create_event',
+    events: _.extend({}, Dialog.prototype.events, {
+        'change select[name="event_location"]': '_onLocationChanged',
+    }),
+
+    /**
+     * @override
+     * @param {Object} parent
+     * @param {Object} options
+     */
+    init: function (parent, options) {
+        options = _.defaults(options || {}, {
+            title: _t("New Event"),
+            size: 'medium',
+            buttons: [
+                {
+                    text: _t("Create"),
+                    classes: 'btn-primary',
+                    click: this.onCreateClick.bind(this),
+                },
+                {
+                    text: _t("Discard"),
+                    close: true
+                },
+            ]
+        });
+        this._super(parent, options);
+    },
+    start: function () {
+        var self = this;
+        return this._super.apply(this, arguments).then(function () {
+            self.$('div.o_event_form_datetime').each(function () {
+                self._initDateTimePicker($(this));
+            });
+
+            var $input = self.$('#address_id');
+            $input.select2({
+                width: '100%',
+                allowClear: true,
+                formatNoMatches: false,
+                multiple: false,
+                selection_data: false,
+                fill_data: function (query, data) {
+                    var that = this;
+                    var tags = {results: []};
+                    _.each(data, function (obj) {
+                        var display_text = obj.display_name + ': ' + obj.street + ', ' + (obj.zip ? (obj.zip + ', ') : '') + obj.city + ', ' + obj.country_id[1];
```

Comment:
Could be simplified by trying to re-use existing address formatting tools on partner I suppose (see other comment).